src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
+                                    Entry::Vacant(_) => panic!("Must have cap if using record"),
+                                };
+
+                                let mut session = output.session(&**tx_cap_entry.get());
+                                match value.iter().nth(before_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, -1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match value.iter().nth(after_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, 1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match tx_event_count.get_mut(tx_id) {
+                                    Some(count) => {
+                                        *count -= 1;
+                                        if *count == 0 {
+                                            let (_, tx_cap) = tx_cap_entry.remove_entry();
+                                            binary_cap.downgrade(&tx_cap.time());
+                                        }
+                                    }
+                                    None => panic!("need event count for tx_id {:?}", tx_id),
+                                }
+                            }
+                        }
+
+                        if !data_buffer.is_empty() || !tx_cap_map.is_empty() {
+                            activator.activate_after(Duration::from_secs(1));
+                        } else {
+                            let tx_frontier = tx_metadata_input.frontier().frontier().to_owned();
+                            let data_frontier = data_input.frontier().frontier().to_owned();
+                            let meet = data_frontier.meet(&tx_frontier);
+                            // Having binary_cap be Capability<u64> rather than Capability<Antichain<u64>>
+                            // means we can't just clear it. Other choice is wrapping it in an `Option` and
+                            // then `take`ing it if `meet` is the empty Antichain. But then we'd have a
+                            // bunch of unwraps everywhere.
+                            let meet_u64 = meet.into_iter().next().unwrap_or(u64::MAX);
+                            let _ = binary_cap.try_downgrade(&meet_u64);
```

Comment:
Following from the comment further up, we should just leave `binary_cap` alone to get dropped during initialization and work solely with the capabilities of the inputs